### PR TITLE
detect the suite from a test path relative to the current working dir

### PR DESCRIPTION
**Pull request by ctrl-f5**
Detect the suite from a test file or path relative to the current working directory.

This fixes the issue that the suite cant be detected when `codeception.yml` is in any subdirectory, and subsequently also fixes the issue when it's inside the tests directory, for which there was a single use case fix in place.

### CODE REVIEW

**Comment by samdark** [comment]
@Naktibalda worth merging master into this one and checking test results with GitHub actions.

**Comment by ctrl-f5** [comment]
@samdark updated with upstream, seems to run fine in the new actions setup!

**Comment by samdark** [comment]
It doesn't break current tests. That's good. It would be great to add some tests verifying the change.

**Comment by ctrl-f5** [comment]
It was basically already tested by `CodeceptionYmlInTestsDirCest` since that's one use case. I've also added a more convoluted test case with some more directories to traverse, based on how we use it in our projects.

File: src/Codeception/Command/Run.php [hunk 1]
```diff
@@ -443,6 +436,9 @@ protected function matchSingleTest($suite, $config)
                 }
                 $testsPath = $config['paths']['tests'] . DIRECTORY_SEPARATOR . $suiteConfig['path'];
                 if ($suiteConfig['path'] === '.') {
+                    if ($config['paths']['tests'] === '.') {
+                        return ['', $s, $suite];
```
Comment (Naktibalda) [inline src/Codeception/Command/Run.php:440]: I don't like this line at all. 

This code is inside `matchSingleTest` method and the third element in result array is supposed to be a path to specific test file.
This condition doesn't even check if `$suite` is a path.

Comment (ctrl-f5) [reply src/Codeception/Command/Run.php:440]: The type of check you describe was not there in the past, nor in this function (line 444), nor in `matchTestFromFilename`. They both just do a `preg_match` on the input and return the result.

`$suite` is never validated as a path in any other code branch.

Comment (ctrl-f5) [reply src/Codeception/Command/Run.php]: anyway, I've removed this code and put back the original hack at the start of the function

**Comment by Naktibalda** [comment]
@ctrl-f5 Your change broke `codecept run tests` - it is fails with this error:
```
In Run.php line 545:

  Test file can't be matched
```
#6103

Could you fix it?

**Comment by Naktibalda** [comment]
I implemented patch, which will make your change to apply only if relative path is inside tests directory #6105

**Pull request merged**